sdk/identity/azure-identity/inc/azure/identity/client_certificate_credential.hpp

```diff
@@ -114,7 +124,25 @@ namespace Azure { namespace Identity {
         = Core::Credentials::TokenCredentialOptions());
 
     /**
-     * @brief Constructs a Client Secret Credential.
+     * @brief Constructs a Client Certificate Credential.
+     *
+     * @param tenantId Tenant ID.
+     * @param clientId Client ID.
+     * @param clientCertificate The x509 certificate which is used for signing, in base64 string
+     * format, including the begin and end headers.
+     * @param privateKey The binary representation of the corresponding RSA private key of the
+     * certificate.
+     * @param options Options for token retrieval.
+     */
+    explicit ClientCertificateCredential(
+        std::string tenantId,
+        std::string const& clientId,
+        std::vector<uint8_t> clientCertificate,
+        std::vector<uint8_t> privateKey,
```

Comment:
What is the format of the private key? Why is the certificate base64 encoded but the key is not base64 encoded?

Reply:
As discussed offline, I will turn this into PEM-encoded format (base64 encoded std::string). And get customer feedback.

My initial intent to have the private key be a DER-encoded binary format, but we can wait to see what the end user expects here.

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-cpp/pull/5991